eth-repository/src/main/kotlin/net/consensys/cava/eth/repository/BlockchainRepository.kt

```diff
@@ -0,0 +1,241 @@
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the bytes if found
+   */
+  suspend fun retrieveBlockBytes(blockHash: Hash): Bytes? {
+    return retrieveBlockBytes(blockHash.toBytes())
+  }
+
+  /**
+   * Retrieves a block into the repository as its serialized RLP bytes representation.
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the bytes if found
+   */
+  suspend fun retrieveBlockBytes(blockHash: Bytes): Bytes? {
+    return blockStore.get(blockHash)
+  }
+
+  /**
+   * Retrieves a block into the repository.
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the block if found
+   */
+  suspend fun retrieveBlock(blockHash: Hash): Block? {
+    return retrieveBlock(blockHash.toBytes())
+  }
+
+  /**
+   * Retrieves a block into the repository.
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the block if found
+   */
+  suspend fun retrieveBlock(blockHash: Bytes): Block? {
+    val bytes: Bytes? = retrieveBlockBytes(blockHash)
+    return if (bytes == null) {
+      null
+    } else {
+      Block.fromBytes(bytes)
+    }
+  }
+
+  /**
+   * Retrieves a block header into the repository as its serialized RLP bytes representation.
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the block header bytes if found
+   */
+  suspend fun retrieveBlockHeaderBytes(blockHash: Hash): Bytes? {
+    return retrieveBlockBytes(blockHash.toBytes())
+  }
+
+  /**
+   * Retrieves a block header into the repository as its serialized RLP bytes representation.
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the block header bytes if found
+   */
+  suspend fun retrieveBlockHeaderBytes(blockHash: Bytes): Bytes? {
+    return blockHeaderStore.get(blockHash)
+  }
+
+  /**
+   * Retrieves a block header into the repository.
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the block header if found
+   */
+  suspend fun retrieveBlockHeader(blockHash: Hash): BlockHeader? {
+    val bytes = retrieveBlockHeaderBytes(blockHash.toBytes()) ?: return null
+    return BlockHeader.fromBytes(bytes)
+  }
+
+  /**
+   * Retrieves a block header into the repository.
+   *
+   * @param blockHash the hash of the block stored
+   * @return a future with the block header if found
+   */
+  suspend fun retrieveBlockHeader(blockHash: Bytes): BlockHeader? {
+    val bytes = retrieveBlockHeaderBytes(blockHash) ?: return null
+    return BlockHeader.fromBytes(bytes)
+  }
+
+  /**
+   * Retrieves the block identified as the chain head
+   *
+   * @return the current chain head, or the genesis block if no chain head is present.
+   */
+  suspend fun retrieveChainHead(): Block? {
+    val bytes = chainMetadata.get(Bytes.wrap("chainHead".toByteArray(StandardCharsets.UTF_8)))
```

Comment:
Would it be better to use an enumeration rather than literal strings for these keys?